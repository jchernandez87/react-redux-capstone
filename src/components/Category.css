.cardsContainer {
  display: grid;
  grid-template-columns: 1fr 1fr;
  background-color: #fa4f91;
}

.flexColumn {
  display: flex;
  flex-direction: column;
}

.header {
  color: white;
  justify-content: center;
  align-items: flex-end;
  padding: 18px;
  background-color: #fa4f91;
}

.header h2 {
  font-size: 24px;
  font-weight: 600;
}

.header span {
  font-size: 14px;
  font-weight: 600px;
}
