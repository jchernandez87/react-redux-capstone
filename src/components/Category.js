import './Category.css';
import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchData } from '../redux/CategorySlice';
import Card from './CategoryCard';

const Category = () => {
  const dispatch = useDispatch();
  const countryName = useSelector((state) => state.categories.countryName);
  const countryCases = useSelector((state) => state.categories.todayCases);
  const categoriesStatus = useSelector((state) => state.categories.status);
  const categories = useSelector((state) => state.categories.list);

  useEffect(() => {
    if (categoriesStatus === 'idle') {
      dispatch(fetchData());
    }
  }, []);

  return (
    <div className="container">
      <div className="header flexColumn">
        <div className="infoContainer">
          <h2 className="lato">{countryName}</h2>
          <span className="lato">{countryCases}</span>
        </div>
      </div>
      <div className="byRegion lato">
        <span>STATS BY REGION</span>
      </div>
      <div className="cardsContainer">
        {categories.map((el) => (
          <Link key={el.id} to={`/details/${el.id}`} className="row">
            <Card id={el.id} city={el.name} activeCases={el.confirmedToday} />
          </Link>
        ))}
        <div className="empty" />
      </div>
    </div>
  );
};

export default Category;
