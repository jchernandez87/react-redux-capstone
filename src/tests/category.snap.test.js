import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Category from '../components/Category';
import store from '../redux/Store';

it('Display category element', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Category />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
