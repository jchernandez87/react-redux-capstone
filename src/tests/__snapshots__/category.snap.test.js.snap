// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display category element 1`] = `
<div
  className="container"
>
  <div
    className="header flexColumn"
  >
    <div
      className="infoContainer"
    >
      <h2
        className="lato"
      >
        SPAIN
      </h2>
      <span
        className="lato"
      >
        0
      </span>
    </div>
  </div>
  <div
    className="byRegion lato"
  >
    <span>
      STATS BY REGION
    </span>
  </div>
  <div
    className="cardsContainer"
  >
    <div
      className="empty"
    />
  </div>
</div>
`;
